File: PropertyManagementSystem/PropertyManagementSystem/Models/MaintenanceAgreement.cs
using System;

namespace PropertyManagementSystem.Models
{
    /// <summary>
    /// Maintenance Agreement between Owners & Association
    /// </summary>
    public class MaintenanceAgreement:BaseModel
    {
        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public long ApartmentId { get; set; }
        public Apartment Apartment { get; set; }
        public DateTime FromDate { get; set; }
        public DateTime ToDate { get; set; }        
        public float MaintenanceAmount { get; set; }
    }
}

File: PropertyManagementUI/src/app/Models/MaintenanceAgreement.ts
import { Apartment } from "./Apartment";
import { AppUserDto } from "./AppUser";
import { BaseModel } from "./BaseModel";

export interface MaintenanceAgreement extends BaseModel{
  AppUserId:string,
  AppUser: AppUserDto | null,
  ApartmentId: number,
  Apartment: Apartment | null,
  FromDate:Date,
  ToDate: Date,
  MaintenanceAmount:number,
  IsDisabled: boolean
}
